week10/pages/Detail.jsx

```diff
@@ -0,0 +1,159 @@
+import { useState, useEffect } from "react";
+import { PostDetail, EditPost, DeletePost } from "../api/post";
+import { PutHeart, DeleteHeart, CntHeart } from "../api/heart";
+import { CreateComment, DeleteComment, GetComment } from "../api/comment";
+import styled from "styled-components";
+import { useParams, useNavigate } from "react-router-dom";
+const Title = styled.div``;
+const Content = styled.div``;
+const Image = styled.img`
+  height: 300px;
+  width: 300px;
+`;
+const CommentList = styled.div``;
+const CommentItem = styled.div``;
+const Detail = () => {
+  const { postId } = useParams();
+  const [post, setPost] = useState(null);
+  const [likeCnt, setLikeCnt] = useState(0);
+  const [comment, setComment] = useState();
+  const [comments, setComments] = useState([]);
+  const [commentCnt, setCommentCnt] = useState(0);
+  const [isEditing, setIsEditing] = useState(false);
+  const [editedTitle, setEditedTitle] = useState("");
+  const [editedContent, setEditedContent] = useState("");
+  const navigate = useNavigate();
+  useEffect(() => {
+    const fetchPostDetail = async () => {
+      try {
+        const data = await PostDetail(postId);
+        setPost(data);
+        setEditedTitle(data.title);
+        setEditedContent(data.content);
+        const heartData = await CntHeart(postId);
+        setLikeCnt(heartData);
+        const commentData = await GetComment(postId);
+        setComments(commentData);
+        setCommentCnt(commentData.length);
+      } catch (err) {
+        console.log(err);
+      }
+    };
+    fetchPostDetail();
+  }, [postId]);
+  const handleEdit = async () => {
+    try {
+      await EditPost(postId, { title: editedTitle, content: editedContent });
+      setIsEditing(false);
+      const data = await PostDetail(postId);
+      setPost(data);
+    } catch (err) {
+      console.log(err);
+    }
+  };
+  const handleDelete = async () => {
+    try {
+      await DeletePost(postId);
+      navigate("/");
+    } catch (err) {
+      console.log(err);
+    }
+  };
+  if (!post) {
+    return <div>Loading...</div>;
+  }
+  const like = async () => {
+    try {
+      await PutHeart(postId);
+      setLikeCnt((prev) => prev + 1);
+    } catch (err) {
+      console.log(err);
+    }
+  };
+  const deleteLike = async () => {
+    try {
+      await DeleteHeart(postId);
+      setLikeCnt((prev) => prev - 1);
+    } catch (err) {
+      console.log(err);
+    }
+  };

```

Comment:
P5: 좋아요와 좋아요 취소를 하나의 handleLike 함수로 작성할 수 있을 것 같아요!

```suggestion
const handleLike = async (isLike) => {
  try {
    if (isLike) {
      await PutHeart(postId);
      setInteraction((prev) => ({ ...prev, likeCount: prev.likeCount + 1 }));
    } else {
      await DeleteHeart(postId);
      setInteraction((prev) => ({ ...prev, likeCount: prev.likeCount - 1 }));
    }
  } catch (err) {
    console.error(err);
  }
};
```